### ISSUE TITLE
Views Datasource JSON data

### ISSUE THREAD

**Issue by Yiphi** [kylebrowning/drupal-ios-sdk#56]
I've been able to successfully login to my Drupal site from within my iOS app--GREAT!!!  Additionally, viewGet successfully runs and returns a view for the currently logged in user only--GREAT!!!  The data in the returned responseObject is JSON output from views datasource and contains data that I want to use to populate tables in the app.

My question is, how do I extract the data from responseObject?  It looks like it's already an array but I don't believe that any key:value pairs have been assigned yet.  As such, do I need to create an NSArray and assign key:value pairs using data from responseObject?  Or, rather, do I need to create an NSDictionary and assign key:value pairs?  Or, do I need to use SBJSON parser to parse the data and then assign key:value pairs into an array or dictionary?  Or, am I thinking about it completely incorrectly?

I apologize for the newbie nature of this question and thank everyone in advance for their assistance.  If it helps, I have copied the views datasource JSON output that is returned by viewGet's responseObject below:

{"myItems":[{"item":{"uid":"number","sid":"number","itemTitle":"text","itemDate":"shortdate","itemSetting":"text"}}]}

Long story short, I want to create table entries in the app that display:

Top half of table row:  "itemDate"
Bottom half of table row:  "itemTitle (itemSetting)"

I will use uid and sid to display userID and submission specific node data later.

Again, thank you all in advance for your assistance and enjoy the rest of your weekend!!

**Comment by Yiphi** [comment]
#kylebrowning #drupal

**Comment by kylebrowning** [comment]
Just as an FYI, im not going to help you build your application and teach you Objective-c, this is something you can answer on stackoverflow as it has nothing to do with the Drupal iOS SDK. 

That being said, response Object from views is always a dictionary.

[responseObject objectForKey:"myItems"]; will give you an array, from there you can iterate through the items and grab the values via the same way

[item objectForKey:@"uid"]; Would give you the UID.

Go read something along these lines and please use this issue queue for Drupal IOS SDK questions only.
http://mobiledevelopertips.com/objective-c/basics-of-objective-c-and-dictionaries-nsdictionary.html

**Comment by Yiphi** [comment]
Thank you so much for your reply, Kyle.

I agree with you 100% and do NOT believe that you (or any other developer for that matter) should teach me objective-c.  I have actually procured several contacts and programming resources so that I can learn it on my own.

That being said, I had a very hard time finding any resource that explained what I needed the way you did in just one sentence.  In fact, I researched the web, stackoverflow, github, drupal.org, and the big nerd ranch textbook.  While they were all useful and helpful in their own right, none of them ever so succinctly stated, "...response Object from views is always a dictionary."  Thus, I didn't have a good idea where to start and posted my question wherever I thought a programmer or user would stumble upon it.  Obviously, it could be that it was always there and I just overlooked it.

In any case, I truly appreciate you making the exception to point me in the right direction.  I apologize for posting my objective-c question in the Drupal iOS SDK issue queue--I assure you it won't happen again.

Again, thank you for all your dedication and help.  Take care and have a great day!
